Users: Adds create user page to users section

jenny-s51 opened the pull request:
## Motivation
<!-- Add references to relevant tickets, issues, design specs and/or a short description of what motivated you to do it. -->

- https://issues.redhat.com/browse/APPDUX-860
- Towards https://issues.redhat.com/browse/APPDUX-861

## Brief Description
<!-- Add a short answer for: 
What was done in this PR? (e.g Fix to prevent users from accessing feature X.)
Why it was done? (e.g Feature X was deprecated.)
-->
This PR adds a "Create User" page to the Users section of Keycloak. 

## Verification Steps
<!--
Add the steps required to verify this change. Keep in mind that these steps should be written so groups unfamiliar with the 
new functionality can follow them, such as QE or documentation.-->

1. Go to Users section of Keycloak.
2. Click "Add User".
3. Verify the page matches the mock: https://marvelapp.com/prototype/7i7ja65/screen/75693207
4. Toggle help and confirm that the text content matches what is expected.
5. Click "Cancel" and make sure you are navigated back to the initial users page.
6. Click the "Add User" button again and fill in the Username.
7. Go back to the table and verify that the new user is in the table.

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [x] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

## Additional Notes
<!-- 
Add images and/or screen caps to illustrate what was changed if this pull request adds to or modifies existing user-visible appearance/output. 
-->

edewit reviewed (approved):
works and looks great, no need to only have it as draft IMO

mfrances17 reviewed (changes requested):
When I navigate to the user page, the 'Create new user' link does not work, because I'm on the 'Start by searching for users' page:
![search-for-users](https://user-images.githubusercontent.com/39063664/109980443-ea8a3c80-7ccd-11eb-9843-30add3f2e5a9.png)

You should add the call to that link. 

I also was not able to navigate if the user was not found:
![no-users](https://user-images.githubusercontent.com/39063664/109980492-f7a72b80-7ccd-11eb-80b6-a66bff95ede3.png)
You should add the call to that button as well.

I was able to access the Create user form via searching for a username that was valid, or by navigating to the URI directly.

mfrances17 commented on src/user/UserForm.tsx:
```diff
@@ -0,0 +1,244 @@
+        />
+      </FormGroup>
+
+      <FormGroup
```
i was able to add invalid email... this should do some basic validation for valid email address

jenny-s51 replied:
nice catch - updated!

mfrances17 commented on src/user/UsersTabs.tsx:
```diff
@@ -0,0 +1,45 @@
+  const save = async (user: UserRepresentation) => {
+    try {
+      await adminClient.users.create({ username: user!.username });
+
```
On successful create, should navigate the user back to main user page with the new user name in the table. Otherwise, can do multiple creates of same username.

mfrances17 commented on cypress/support/pages/admin_console/manage/users/CreateUserPage.ts:
```diff
@@ -0,0 +1,41 @@
+    return this;
+  }
+
+  goToCreateUser() {
```
this fails for me... my users page is empty and displays the 'Start by searching for users' search form

jenny-s51 commented:
@mfrances17 thank you for your helpful review comments! I've addressed your feedback and made the corresponding updates to the UI and the cypress tests - please let me know if you're seeing the tests pass now 🙂

mfrances17 reviewed (approved):
LGTM, looks like all of my suggestions/comments were addressed. And tests all pass. Good job!
